### DISCUSSION TITLE
Displaying tikz diagrams

### DISCUSSION THREAD

**Discussion by quarkQuark** [org-roam/org-roam-ui#270] [Q&A]
Org-roam-ui seems unable to display tikz (or at least tikz-cd) diagrams. From a little digging, it seems that this is a limitation of KaTeX. Does anyone know of a workaround, perhaps by switching the rendering engine or interfacing with fragment previews? My notes make very heavy use of commutative diagrams, so I can't use org-roam-ui otherwise.

**Comment by splintersuidman** [comment]
It seems KaTeX has (limited) support for drawing commutative diagrams in a `cd` environment (see KaTeX/KaTeX#2396). But as far as I can tell, you cannot draw diagonal or bent arrows.

I came up with the following way of including commutative diagrams drawn using `tikzcd` in org-roam-ui, partially based on [this blog post](https://fanpengkong.com/post/org-graphics/):
```org
#+header: :headers '("\\usepackage{tikzcd}")
#+header: :imagemagick yes :iminoptions -density 600 :imoutoptions -geometry 400
#+header: :fit yes :noweb yes
#+begin_src latex :exports results :results output graphics file :file test.png
  \begin{tikzcd}
    X \ar[r, "\eta_X"] \ar[d, "f"'] & UFX \ar[d, "UFf"] \\
    Y \ar[r, "\eta_Y"'] & UFY
  \end{tikzcd}
#+end_src
```
Then you can use `C-c C-c` on the code block to compile the image, and it will be included in the file.
(You might need to evaluate `(org-babel-do-load-languages 'org-babel-load-languages '((latex . t)))` for this to work. You also need the `imagemagick` program installed, which is used for the conversion to PNG.)

In the org-roam-ui interface, it looks like this (I also loaded the `libertine` font package):
![Screenshot_20230220_095237](https://user-images.githubusercontent.com/16049456/220057986-8a3137d5-950e-4091-aa05-6b99faf8a55e.png)

I have not yet figured out how to prevent the code block from showing up in the org-roam-ui interface. I guess org-roam-ui does not use org-export, so the `:exports results` option does not work.

**Comment by PatrickNorton** [comment]
I wonder if it would be possible to modify org-roam-ui to use something like [tikzjax](https://github.com/kisonecat/tikzjax) to display TikZ.
